src/libaktualizr/primary/aktualizr.cc

```diff
@@ -83,14 +83,32 @@ bool Aktualizr::UptaneCycle() {
 std::future<void> Aktualizr::RunForever() {
   std::future<void> future = std::async(std::launch::async, [&]() {
     SendDeviceData().get();
-    while (UptaneCycle()) {
-      std::this_thread::sleep_for(std::chrono::seconds(config_.uptane.polling_sec));
+
+    std::unique_lock<std::mutex> l(exit_cond_.m);
+    while (true) {
+      if (!UptaneCycle()) {
+        break;
+      }
+
+      if (exit_cond_.cv.wait_for(l, std::chrono::seconds(config_.uptane.polling_sec),
+                                 [this] { return exit_cond_.flag; })) {
+        break;
+      }
     }
     uptane_client_->completeInstall();
```

Comment:
If completeInstall() is not called will it be properly handled during the subsequent Aktualizr/UptaneCycle() ?

Reply:
Right now this method is used for rebooting. So:

- if the system reboots and aktualizr is killed before this method, nothing is lost
- if aktualizr receives SIGKILL or crashes before this method, `isInstallCompletionRequired()` should still return true when systemd would restart aktualizr and it should reboot then. But I don't think we cover that with any unit test.